torchaudio/models/_wavernn.py

```diff
@@ -103,3 +105,122 @@ def forward(self, x: Tensor) -> Tensor:
         """
 
         return self.melresnet_model(x)
+
+
+class _Stretch2d(nn.Module):
+    r"""This is a two-dimensional stretch layer. It is a block used in WaveRNN. WaveRNN is based on the paper
+    "Efficient Neural Audio Synthesis". Nal Kalchbrenner, Erich Elsen, Karen Simonyan, Seb Noury,
+    Norman Casagrande, Edward Lockhart, Florian Stimberg, Aaron van den Oord, Sander Dieleman,
+    Koray Kavukcuoglu. arXiv:1802.08435, 2018.
+
+    Args:
+        x_scale: the scale factor in x axis (required)
+        y_scale: the scale factor in y axis (required)
+
+    Examples::
+        >>> stretch2d = _Stretch2d(x_scale=1, y_scale=1)
+
+        >>> input = torch.rand(10, 1, 100, 512)
+        >>> output = stretch2d(input)
+    """
+
+    def __init__(self,
+                 x_scale: int,
+                 y_scale: int) -> None:
+        super().__init__()
+
+        self.x_scale = x_scale
+        self.y_scale = y_scale
+
+    def forward(self, x: Tensor) -> Tensor:
+        r"""Pass the input through the _Stretch2d layer.
+
+        Args:
+            x: the input sequence to the _Stretch2d layer (required)
+
+        Shape:
+            - x: :math:`(batch_size, channel, freq, time)`
+            - output: :math:`(batch_size, channel, freq * y_scale, time * x_scale)`
+        """
+
+        batch_size, channel, freq, time = x.size()
+        x = x.unsqueeze(-1).unsqueeze(3)
+        x = x.repeat(1, 1, 1, self.y_scale, 1, self.x_scale)
+        return x.view(batch_size, channel, freq * self.y_scale, time * self.x_scale)
+
+
+class _UpsampleNetwork(nn.Module):
+    r"""This is an upsample block based on a stack of Conv2d and Strech2d layers.
+    It is a block used in WaveRNN. WaveRNN is based on the paper "Efficient Neural Audio Synthesis".
+    Nal Kalchbrenner, Erich Elsen, Karen Simonyan, Seb Noury, Norman Casagrande, Edward Lockhart,
+    Florian Stimberg, Aaron van den Oord, Sander Dieleman, Koray Kavukcuoglu. arXiv:1802.08435, 2018.
+
+    Args:
+        upsample_scales: the list of upsample scales (required)
```

Comment:
nit: remove "(required)"

Reply:
All "(required)"s in docstring have been removed.